### PR TITLE
Fixed NNP isolator test on systems with POSIX-compliant /bin/sh.

### PR DESCRIPTION
**Pull request by cf-natali**
The test used some non-POSIX features such as arrays when parsing
/proc/self/status, which breaks on systems where /bin/sh is
POSIX-compliant, e.g. on Debian which uses dash.

See example failure:
```
[ RUN      ] LinuxNNPIsolatorTest.ROOT_CheckNoNewPrivileges
sh: 1: hadoop: not found
Preparing rootfs at '/tmp/LinuxNNPIsolatorTest_ROOT_CheckNoNewPrivileges_NUOIDx/provisioner/containers/b02db6ab-df16-4439-812a-0f6379c754b9/backends/copy/rootfses/5fd35417-bdbf-4303-90f9-b419c34fb25e'
Marked '/' as rslave
Changing root to /tmp/LinuxNNPIsolatorTest_ROOT_CheckNoNewPrivileges_NUOIDx/provisioner/containers/b02db6ab-df16-4439-812a-0f6379c754b9/backends/copy/rootfses/5fd35417-bdbf-4303-90f9-b419c34fb25e
sh: 5: Syntax error: redirection unexpected
../../src/tests/containerizer/linux_nnp_isolator_tests.cpp:137: Failure
Value of: WEXITSTATUS(wait->get().status())
  Actual: 2
Expected: 0
Which is: 0
[  FAILED  ] LinuxNNPIsolatorTest.ROOT_CheckNoNewPrivileges (1102 ms)
```

### CODE REVIEW

File: src/tests/containerizer/linux_nnp_isolator_tests.cpp [hunk 1]
```diff
@@ -102,18 +102,26 @@ TEST_F(LinuxNNPIsolatorTest, ROOT_CheckNoNewPrivileges)
...
   ExecutorInfo executor = createExecutorInfo(
       "test_executor",
       R"~(
-      #!/bin/bash
-      x=$(cat /proc/self/status);
-      y=${x##*NoNewPrivs:};
-      read -a a <<< $y;
-      if [ ${a[0]} == "1" ]; then exit 0; else exit 1; fi
+      nnp_seen="false"
+      for line in $(cat /proc/self/status); do
```
Comment (asekretenko) [inline src/tests/containerizer/linux_nnp_isolator_tests.cpp:113]: @cf-natali IMO, `line` is a very unfortunate name for this variable; readers can be mislead into thinking that lines in `/proc/self/status` contain no whitespaces and this script reads it line by line.
Something like `word`/`piece`/`segment`/... would be much less confusing.

**Comment by cf-natali** [comment]
Ping?

**Comment by cf-natali** [comment]
@qianzhangxa @asekretenko 
Since you mentioned being still interested in the project it'd be great if you could review :).

**Comment by asekretenko** [comment]
Looks good to me; merged. Thanks for finding and fixing this!

**Pull request merged**